pkg/controller/directvolumemigration/rsync.go

```diff
@@ -2450,11 +2456,11 @@ func (t *Task) getLatestPodForOperation(client compat.Client, operation migapi.R
 		} else if _, err := strconv.Atoi(val); err != nil {
 			continue
 		}
-		if pod.CreationTimestamp.After(mostRecentPod.CreationTimestamp.Time) {
-			mostRecentPod = pod
+		if mostRecentPod != nil && pod.CreationTimestamp.After(mostRecentPod.CreationTimestamp.Time) {
```

Comment:
This branch will never be reached because, mostRecentPod is always nil. 
```suggestion
                if mostRecentPod == nil { 
                      mostRecentPod = &pod
                } else if pod.CreationTimestamp.After(mostRecentPod.CreationTimestamp.Time) {
```